comid/measurement_test.go

```diff
@@ -112,3 +121,216 @@ func TestMeasurement_NewUUIDMeasurement_bad_uuid(t *testing.T) {
 
 	assert.Nil(t, tv.SetUUID(nonRFC4122UUID))
 }
+
+var (
+	testMKeyUintMin uint64 = 0
+	testMKeyUintMax uint64 = ^uint64(0)
+)
+
+func TestMkey_Valid_no_value(t *testing.T) {
+	mkey := &Mkey{}
+	expectedErr := "unknown measurement key type: <nil>"
+	err := mkey.Valid()
+	assert.EqualError(t, err, expectedErr)
+}
+
+func TestMeasurement_MarshalCBOR_uint_mkey_ok(t *testing.T) {
+	tvs := []struct {
+		mkey     uint64
+		expected []byte
+	}{
+		{
+			mkey:     testMKeyUintMin,
+			expected: MustHexDecode(t, "00"),
+		},
+		{
+			mkey:     TestMKey,
+			expected: MustHexDecode(t, "1902BC"),
+		},
+		{
+			mkey:     testMKeyUintMax,
+			expected: MustHexDecode(t, "1BFFFFFFFFFFFFFFFF"),
+		},
+	}
+
+	for _, tv := range tvs {
+		meas := NewUintMeasurement(tv.mkey)
+		require.NotNil(t, meas)
+
+		actual, err := meas.Key.MarshalCBOR()
+		assert.Nil(t, err)
+		assert.Equal(t, tv.expected, actual)
+		fmt.Printf("CBOR: %x\n", actual)
+	}
+}
+func TestMkey_MarshalCBOR_uint_notok(t *testing.T) {
+	tvs := []struct {
+		expected string
+	}{
+		{
+			expected: "unknown measurement key type: <nil>",
+		},
+	}
+
+	for _, tv := range tvs {
+		mkey := &Mkey{}
+		_, err := mkey.MarshalCBOR()
+
+		assert.EqualError(t, err, tv.expected)
+	}
+}
+
+func TestMkey_UnmarshalCBOR_uint_ok(t *testing.T) {
+	tvs := []struct {
+		mkey     []byte
+		expected uint64
+	}{
+		{
+			mkey:     MustHexDecode(t, "00"),
+			expected: testMKeyUintMin,
+		},
+		{
+			mkey:     MustHexDecode(t, "1902BC"),
+			expected: TestMKey,
+		},
+		{
+			mkey:     MustHexDecode(t, "1BFFFFFFFFFFFFFFFF"),
+			expected: testMKeyUintMax,
+		},
+	}
+
+	for _, tv := range tvs {
+		mKey := &Mkey{}
+
+		err := mKey.UnmarshalCBOR(tv.mkey)
+		assert.Nil(t, err)
+		actual, err := mKey.GetKeyUint()
+		assert.Nil(t, err)
+		assert.Equal(t, tv.expected, actual)
+	}
+}
+
+func TestMkey_UnmarshalCBOR_uint_notok(t *testing.T) {
+	tvs := []struct {
+		mkey     []byte
+		expected string
+	}{
+		{
+			mkey:     []byte{0xAB, 0xCD},
+			expected: "unexpected EOF",
+		},
+	}
+
+	for _, tv := range tvs {
+		mKey := &Mkey{}
+
+		err := mKey.UnmarshalCBOR(tv.mkey)
+
+		assert.EqualError(t, err, tv.expected)
+	}
+}
+
+func TestMkey_MarshalJSON_uint_ok(t *testing.T) {
+	tvs := []struct {
+		mkey     uint64
+		expected []byte
+	}{
+		{
+			mkey:     testMKeyUintMin,
+			expected: []byte(`{"type":"uint","value":0}`),
+		},
+		{
+			mkey:     TestMKey,
+			expected: []byte(`{"type":"uint","value":700}`),
+		},
+		{
+			mkey:     testMKeyUintMax,
+			expected: []byte(`{"type":"uint","value":18446744073709551615}`),
+		},
+	}
+
+	for _, tv := range tvs {
+
+		meas := NewUintMeasurement(tv.mkey)
+		require.NotNil(t, meas)
+
+		actual, err := meas.Key.MarshalJSON()
+		assert.Nil(t, err)
+		assert.Equal(t, tv.expected, actual)
+
+		fmt.Printf("JSON: %x\n", actual)
+	}
+}
+
+func TestMkey_MarshalJSON_uint_notok(t *testing.T) {
```

Comment:
Same set of comments as above:
1. we have one only vector therefore we don't need a full-blown table driven test
but most importantly:
1. there is nothing specific to the uint mkey that is tested here

Reply:
For 1. I agree, that I will perhaps add one more test so that it is justified to have a table driven logic.